pkg/validate/raw/validate/source_spec_validator.go

```diff
@@ -432,3 +419,12 @@ func MissingConfigMapKey(o client.Object, name, key string) status.Error {
 		Sprintf("%s field spec.helm.valuesFileRefs.valuesFile must specify an existing data key in the referenced ConfigMap; data key %q not found in ConfigMap %q", kind, key, name).
 		BuildWithResources(o)
 }
+
+// ConfigMapMustBeImmutable reports that a referenced ConfigMap from RSync spec.helm.valuesFileSources is
+// not immutable.
+func ConfigMapMustBeImmutable(o client.Object) status.Error {
+	kind := o.GetObjectKind().GroupVersionKind().Kind
+	return invalidSyncBuilder.
+		Sprintf("%ss must only reference ConfigMaps that are immutable", kind).
```

Comment:
can we make this message more specific to helm valuesFileRefs? and include the name/ns of the COnfigMap that wasn't immutable

Reply:
done